pennylane/measurements/probs.py

```diff
@@ -264,6 +264,10 @@ def process_density_matrix(self, density_matrix: TensorLike, wire_order: Wires):
             )
 
         # Since we only care about the probabilities, we can simplify the task here by creating a 'pseudo-state' to carry the diagonal elements and reuse the process_state method
+        # Introduce a small epsilon to avoid sqrt(0) to ensure absolute positivity.
+        prob = qml.math.real(prob)
+        prob = qml.math.where(prob < 0, -prob, prob)
+        prob = qml.math.where(prob == 0, prob + np.finfo(float).eps, prob)
         p_state = qml.math.sqrt(prob)
```

Comment:
Let's try

Reply:
Since this involves some temporarily incompatibility between isnan and different interfaces tensor type&shape, let's roll back adn mark this as resolve >.<